File: Problemset/rearrange-spaces-between-words/rearrange-spaces-between-words.cpp

// @Title: 重新排列单词间的空格 (Rearrange Spaces Between Words)
// @Author: Singularity0909
// @Date: 2020-09-20 20:02:48
// @Runtime: 0 ms
// @Memory: 6.1 MB

class Solution {
public:
    string reorderSpaces(string text) {
        int cnt = 0;
        for (char c : text) {
            if (c == ' ') {
                cnt++;
            }
        }
        stringstream ss(text);
        string str, res;
        vector<string> vec;
        while (ss >> str) vec.push_back(str);
        if (vec.size() == 1) return vec.front() + string(cnt, ' ');
        int gap = cnt / (vec.size() - 1);
        int rem = cnt - gap * (vec.size() - 1);
        for (int i = 0; i < (int)vec.size(); i++) {
            res += vec[i] + string(i == vec.size() - 1 ? rem : gap, ' ');
        }
        return res;
    }
};

File: Problemset/rearrange-spaces-between-words/rearrange-spaces-between-words.py

# @Title: 重新排列单词间的空格 (Rearrange Spaces Between Words)
# @Author: Singularity0909
# @Date: 2020-09-23 11:20:25
# @Runtime: 52 ms
# @Memory: 13.4 MB

class Solution:
    def reorderSpaces(self, text: str) -> str:
        cnt = sum([1 if i == ' ' else 0 for i in text])
        print(cnt)
        s = text.split()
        l = len(s)
        print(l)
        print(s)
        if l == 1:
            return s[0] + " " * cnt
        t = (" " * (cnt // (l - 1))).join(s)
        return t + " " * (cnt % (l - 1))
